staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/appendixa_test.go

```diff
@@ -289,11 +289,11 @@ func TestAppendixA(t *testing.T) {
 			},
 		},
 		{
-			example: hex("d74401020304"),
-			decoded: "\x01\x02\x03\x04",
-			encoded: hex("4401020304"),
+			example: hex("d74401020304"), // 23(h'01020304')
+			decoded: "01020304",
+			encoded: hex("483031303230333034"), // '01020304'
```

Comment:
can the comment include an explanation for the '48' tag?

```
48(byte-string) "01020304"
```

does this mean that a base64-encoded byte string doesn't round-trip through cbor back out to cbor that indicates the expected format is base64?

Reply:
The 0x48 isn't considered a tag here, it's the head of a byte string (https://www.rfc-editor.org/rfc/rfc8949.html#section-3-2). Three bits indicate the byte string type and because it is small, the length (8) is encoded in the remaining 5 bits.

Reply:
ok... I can see why this test is structured as-is. Let's add a new unit test that specifically exercises the typed encoding → untyped conversion → typed decoding path and captures the serialized form of encoded byte slices with hints